packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadControls.tsx

```diff
@@ -232,9 +223,11 @@ export const UploadControls = ({
     };
   }, [dispatchStoreAction]);
 
-  const [tasks, handleProcess] = useProcessTasks(uploadHandler, files, {
-    concurrency: 4,
-  });
+  const [{ tasks, isProcessing }, handleProcess] = useProcessTasks(
+    uploadHandler,
+    files,
```

Comment:
Temporary workaround to make the order of the tasks match the sort order. Tested this out and it makes the upload progress order make sense.
```suggestion
    files?.sort(({ key: key1 }, { key: key2 }) => compareStrings(key1, key2)),
```

Reply:
Can we do the sorting on the returned values? Would prefer not to mess with them before they get added as tasks

Reply:
I don't think so because if you don't sort before creating the tasks, then the order they will execute in will not match the table order. That's why the UI upload behavior feels so random, because it's the order the tasks were originally created in.

Reply:
Cool, we can add it as a follow up. MIght make sense to just handle the sorting in the reducer